.pre-commit-config.yaml

```diff
@@ -0,0 +1,29 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+fail_fast: true
+minimum_pre_commit_version: '0'
+repos:
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.4.0
+    hooks:
+    -   id: check-added-large-files
+    -   id: check-case-conflict
+    -   id: check-json
+    # -   id: check-yaml # skip for now
+    -   id: end-of-file-fixer
+    -   id: requirements-txt-fixer
+    -   id: trailing-whitespace
```

Comment:
I'm curious if some of these hooks overlap with the functionality of flake8/black, for example flake8 looks like it already has a [rule](https://www.flake8rules.com/rules/W291.html) for trailing whitespaces.